### PR TITLE
Gauge

### LINKED ISSUE #2: support gauge

### PR DESCRIPTION
**Pull request by overvenus**
Resolved #2 

PTAL @siddontang @disksing @BusyJay

### CODE REVIEW

File: src/gauge.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+/// Desc, but have different values for their variable labels. This is used if
+/// you want to count the same thing partitioned by various dimensions
+/// (e.g. number of operations queued, partitioned by user and operation type).
+pub type GaugeVec = MetricVec<GaugeVecBuilder>;
+
+impl GaugeVec {
+    pub fn new(opts: Opts, label_names: &[&str]) -> Result<GaugeVec> {
+        let variable_names = label_names.iter().map(|s| (*s).to_owned()).collect();
+        let desc = try!(Desc::new(opts.fq_name(), opts.help, variable_names, opts.const_labels));
+        let metric_vec = MetricVec::create(desc, proto::MetricType::COUNTER, GaugeVecBuilder {});
```
Comment (siddontang) [inline src/gauge.rs]: MetricType::GAUGE

File: src/encoder.rs [hunk 2]
```diff
@@ -71,8 +71,12 @@ impl Encoder for TextEncoder {
                     MetricType::COUNTER => {
                         try!(write_sample(name, m, "", "", m.get_counter().get_value(), writer));
                     }
-                    MetricType::GAUGE | MetricType::SUMMARY | MetricType::HISTOGRAM |
-                    MetricType::UNTYPED => unimplemented!(),
+                    MetricType::GAUGE => {
+                        try!(write_sample(name, m, "", "", m.get_gauge().get_value(), writer));
```
Comment (siddontang) [inline src/encoder.rs]: add Guage test for encoder too.

**Comment by siddontang** [comment]
Rest LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**